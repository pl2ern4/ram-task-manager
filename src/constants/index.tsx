const USER = ['Alina Jacob', 'Daniel Bens', 'Jina Thomas','Not Assigned'];

const TICKET_TYPE = ['Story', 'Issue', 'Maintaince','Not Assigned'];

const TICKET_STATUS = ['Todo', 'inProgress', 'Done',];

const TICKET_ORDERBY = {
    modifiedDate: 'MODIFIED_DATE',
    createdDate: 'CREATED_DATE',
    bookmarked: 'BOOKMARKED',
    title: 'TITLE'
};

const ALERT_CONSTANT = {
    story: 'success',
    bug: 'error',
    maintaince:'info'
  }

export {USER, TICKET_TYPE, TICKET_STATUS, TICKET_ORDERBY, ALERT_CONSTANT}