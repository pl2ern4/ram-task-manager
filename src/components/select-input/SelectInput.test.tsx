import * as React from 'react';
import { render } from '@testing-library/react';
import SelectInput from './SelectInput';
import * as constants from '../../constants';

describe('SelectInput', () => {
    it('should Match Snapshot', async () => {
        const { container } = render(<SelectInput label="test" options={constants.TICKET_TYPE} defaultOption={constants.TICKET_TYPE[1]}/>)
        expect(container).toMatchSnapshot();
    });
    it('should Match Snapshot, when `defaultOption` not passed', async () => {
        const { container } = render(<SelectInput label="test" options={constants.TICKET_TYPE} defaultOption={null}/>)
        expect(container).toMatchSnapshot();
    });   
})