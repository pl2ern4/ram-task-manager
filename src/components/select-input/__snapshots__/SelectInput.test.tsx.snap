// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput should Match Snapshot 1`] = `
<div>
  <div
    class="css-1dw64bn"
  >
    <label
      class="css-14pxzl7"
    >
      test
    </label>
    <button
      aria-controls=":r0:"
      aria-expanded="false"
      class="base-Select-root css-12xgkxt"
      role="combobox"
      type="button"
    >
      Issue
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="UnfoldMoreRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
        />
      </svg>
    </button>
    <input
      aria-hidden="true"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      tabindex="-1"
      value="Issue"
    />
  </div>
</div>
`;

exports[`SelectInput should Match Snapshot, when \`defaultOption\` not passed 1`] = `
<div>
  <div
    class="css-1dw64bn"
  >
    <label
      class="css-14pxzl7"
    >
      test
    </label>
    <button
      aria-controls=":r5:"
      aria-expanded="false"
      class="base-Select-root css-12xgkxt"
      role="combobox"
      type="button"
    >
      Story
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="UnfoldMoreRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
        />
      </svg>
    </button>
    <input
      aria-hidden="true"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      tabindex="-1"
      value="Story"
    />
  </div>
</div>
`;
